DeviceManager/DeviceManager.py

```diff
@@ -282,29 +285,68 @@ def update_device(deviceid):
             return format_response(e.error_code, e.message)
 
 
-@device.route('/device/<deviceid>/attrs', methods=['PUT'])
+def find_attribute(orm_device, attr_name, attr_type):
+    """
+    Find a particular attribute in a device retrieved from database.
+    Return the attribute, if found, 'None' otherwise
+    """
+    for template_id in orm_device['attrs']:
+        for attr in orm_device['attrs'][template_id]:
+            if (attr['label'] == attr_name) and (attr['type'] == attr_type):
+                return attr
+    return None
+
+@device.route('/device/<deviceid>/actuate', methods=['PUT'])
 def configure_device(deviceid):
+    """
+    Send actuation commands to the device
+    """
     try:
-        tenant = init_tenant_context(request, db)
-        # In fact, the actual device is not needed. We must be sure that it exists.
-        assert_device_exists(deviceid)
-        json_payload = json.loads(request.data)
+        LOGGER.info('Received request for /device/<id>/actuate.')
+        LOGGER.info('Device ID is: %s', deviceid)
+
+        # Meta information to be published along with device actuation message
+        meta = {
+            'service': ''
+        }
         kafka_handler = KafkaHandler()
-        # Remove topic metadata from JSON to be sent to the device
-        # Should this be moved to a HTTP header?
-        topic = json_payload["topic"]
-        del json_payload["topic"]
+        invalid_attrs = []
+        payload = {}
 
-        kafka_handler.configure(json_payload, meta = { "service" : tenant, "id" : deviceid, "topic": topic})
+        meta['service'] = init_tenant_context(request, db)
 
-        result = {'message': 'configuration sent'}
-        return make_response(result, 200)
+        orm_device = assert_device_exists(deviceid)
+        full_device = serialize_full_device(orm_device)
+        LOGGER.debug('Full device: %s', json.dumps(full_device))
 
-    except HTTPRequestError as e:
-        if isinstance(e.message, dict):
-            return make_response(json.dumps(e.message), e.error_code)
+        payload = json.loads(request.data)
+        LOGGER.debug('Parsed request payload: %s', json.dumps(payload))
+
+        payload['id'] = orm_device.id
+
+        for attr in payload['attrs']:
```

Comment:
I was not sure whether I should add this check here or let it be just in iotagent. I chose to let it here because the user would have a more direct answer that the request was invalid - he/she would not have to receive an alarm or something from iotagent in order to know that the request was wrong.